keymanapp/keyman
# fix(linux): Fix disabling of buttons

When we recently added the new column and changed the model we didn't adjust all the existing references for the model fields.
This change fixes this and also extract some additional methods.

# User Testing

## Preparations

- The tests can be run on any Linux version
- [Install build artifacts of this PR](https://github.com/keymanapp/keyman/wiki/How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux)
- Reboot
- Install Amharic keyboard to shared area:
  ```
  sudo km-package-install --shared --package gff_amharic
  ```
- Install Khmer Angkor keyboard to user area:
  ```
  km-package-install --package khmer_angkor
  ```
- Install EuroLatin (SIL) keyboard for French to user area:
  ```
  km-package-install --package sil_euro_latin  --bcp47 fr
  ```
- Start Keyman Configuration

## Tests

**TEST_UNINST_DISABLED**: Uninstall button disabled for keyboards in Shared area

- In Keyman Configuration select Amharic
- Verify that "Uninstall" and "Options" buttons are grayed out, "About" and "Help" are enabled

**TEST_OPT_ENABLED**: Options button is enabled for keyboards that have options

- In Keyman Configuration select EuroLatin (SIL)
- Verify that all buttons ("Options" "Uninstall", "About", and "Help") are enabled

**TEST_OPT_DISABLED**: Options button is disabled for keyboards that don't have options

- In Keyman Configuration select Khmer Angkor
- Verify that "Options" button is grayed out, "Uninstall", "About", and "Help" are enabled
